core/src/test/java/io/questdb/test/griffin/UpdateTest.java

```diff
@@ -1362,6 +1341,43 @@ public void testUpdateReadonlyFails() throws Exception {
         });
     }
 
+    @Test
+    public void testUpdateRenamedSymbol() throws Exception {
+        Assume.assumeTrue(walEnabled);
+        assertMemoryLeak(() -> {
+            ddl(
+                    "create table test (ts timestamp, x int, y string, sym symbol, symi symbol index) timestamp(ts) partition by DAY WAL"
+            );
+            compile("insert into test select timestamp_sequence('2022-02-24T01:01', 1000000L * 60 * 60), x, 'a', 'abc', 'i' from long_sequence(5)");
+
+            ddl("alter table test add column abc int");
+            ddl("alter table test drop column x");
+            ddl("alter table test rename column y to xxx");
+            ddl("alter table test alter column sym add index");
+            ddl("alter table test dedup enable upsert keys(ts)");
+            ddl("alter table test dedup disable");
+            ddl("alter table test drop partition list '2022-02-23'");
+            ddl("alter table test detach partition list '2022-02-23'");
+            ddl("alter table test attach partition list '2022-02-23'");
+            ddl("alter table test alter column sym cache");
+            ddl("alter table test alter column symi drop index");
+            ddl("alter table test set type bypass wal");
+            update("update test set sym = '2' where sym = '1'");
```

Comment:
was not this test supposed to rename column `sym`?

Reply:
It is enough to drop column `x` to have "reader" column indexes shifted.